docs/GenericsManifesto.md

```diff
@@ -592,6 +622,43 @@ one cannot directly specify `T`: either one calls `f` (and `T` is determined via
 ```Swift
 let x = f<Int> // x has type (Int) -> Void
 ```
+### Conditional default arguments
```

Comment:
Has this feature been discussed previously anywhere? I don't think we should add it to the document. The other two are "canonical" generics features that we have all talked about before.

Reply:
Not to my knowledge.  It was on my mind while I was writing this so I added it.  I can remove it if others agree that it shouldn’t be added.

Reply:
I guess I'd rather have it removed from this PR and discuss it on swift-evolution a little bit first to get some basic syntax ideas in place. And I do want to discuss this... it's something that we might be able to use to help simplify associated type inference.

Reply:
Sounds good.  I'll remove it when I update the PR and send a separate pitch on evolution.

I was thinking of this primarily as a way to reduce the pain of providing an interface like this but if it can help simplify associated type inference it would move up the priority list for me quite a bit.  I have been seeing associated type inference issues in code that uses an overload set like this.